lib/elixir_wallet/key_pair.ex

```
@@ -16,25 +17,50 @@ defmodule KeyPair do
   # Used as guard for the key derivation type: normal / hardned
   @mersenne_prime 2_147_483_647
 
-  def generate_seed(mnemonic, pass_phrase \\ "", opts \\ []) do
-    SeedGenerator.generate(mnemonic, pass_phrase, opts)
+  @doc """
+  Generates a seed from the given mnemonic and pass_phrase
+  """
+  def generate_seed(mnemonic, pass_phrase \\ "") do
+    SeedGenerator.generate(mnemonic, pass_phrase, [])
   end
 
-  ## Generates Bitcoin master key when key type is not specified
-  ## Generates Aeternity master key if :ae is specified as key type
-  def generate_master_key(seed_bin, :seed) do
-    generate_master_key(:crypto.hmac(:sha512, @bitcoin_key, seed_bin), :btc, :private)
+  @doc """
+  Generates master private extended key.
+  ## Examples
+      iex> generate_master_key(seed_bin, :btc)
+      master_extended_btc_key
+
+      iex> generate_master_key(seed_bin, :ae)
+      master_extended_ae_key
+
+  ## Currencies
+     -  `:btc` - Creates a `Bitcoin` key
+     -  `:ae`  - Creates an `Aeternity` key
+  """
+  @spec generate_master_key(Binary.t(), currency::Atom.t()) :: extended_key::Map.t()
+  def generate_master_key(seed_bin, :btc) do
+     build_master_key(:crypto.hmac(:sha512, @bitcoin_key, seed_bin), :btc)
+  end
+  def generate_master_key(seed_bin, :ae) do
+     build_master_key(:crypto.hmac(:sha512, @aeternity_key, seed_bin), :ae)
   end
-  def generate_master_key(seed_bin, :seed, :ae) do
-    generate_master_key(:crypto.hmac(:sha512, @aeternity_key, seed_bin), :ae, :private)
+  def generate_master_key(_, currency) do
+     IO.warn("The cryptocurrency #{currency} is not supported! Check the doc for more info.")
```

maybe throw an exception instead